lib/services/unified_resource_adapter.go

```diff
@@ -0,0 +1,155 @@
+package services
+
+import (
+	"encoding/json"
+	"time"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/utils"
+)
+
+// UnifiedResource153 is a type constraint that requires a type to implement
+// the Resource153 interface AND provides a Clone method.
+type UnifiedResource153[T interface{ CloneResource() T }] interface {
+	types.Resource153
+	CloneResource() T
+}
+
+// UnifiedResource153Adapter is a wrapper around a newer, RFD153-style resource
+// that provides the newer style resources with the interfaces required for use
+// with the Unified resource cache.
+type UnifiedResource153Adapter[T UnifiedResource153[T]] struct {
```

Comment:
The UnifiedResource153Adapter duplicates 80% of the functionality in [Resource153ToLegacy](https://github.com/gravitational/teleport/blob/master/api/types/resource_153.go#L123).

The only missing methods required to make the Unified Resource fit the ResourceWithLabels interface are:

```
func (r Resource) Origin() string {
}

func (r Resource) SetOrigin(s string) {
}

func (r Resource) GetLabel(key string) (value string, ok bool) {
}

func (r Resource) GetAllLabels() map[string]string {
}

func (r Resource) GetStaticLabels() map[string]string {
}

func (r Resource) SetStaticLabels(sl map[string]string) {
}

func (r Resource) MatchSearch(searchValues []string) bool {
}
```


Would it be feasible to simply add these missing methods to [identitycenter.go](https://github.com/gravitational/teleport/blob/master/lib/services/identitycenter.go#L36), or can the existing [Resource153ToLegacy](https://github.com/gravitational/teleport/blob/master/api/types/resource_153.go#L123) implementation be leveraged to avoid introducing additional abstractions and new interfaces for Resource153 and legacy resources?